config/shared_config.go

```diff
@@ -750,13 +761,13 @@ func (c *SharedConfig) setFromIniSections(profiles map[string]struct{}, profile
 		// First time a profile has been seen, It must either be a assume role
 		// or credentials. Assert if the credential type requires a role ARN,
 		// the ARN is also set.
-		if err := c.validateCredentialsRequireARN(profile); err != nil {
+		if err := c.validateCredentialsConfig(profile); err != nil {
 			return err
 		}
 	}
 
 	// if not top level profile and has credentials, return with credentials.
-	if len(profiles) != 0 && c.Credentials.HasKeys() {
+	if len(profiles) != 0 && (c.Credentials.HasKeys()) {
```

Comment:
not important but, wouldn't this be a gofmt fixup?  I'd not expect the wrapping `(`, `)` to be needed.